src/sentry/models/releases/release_project.py

```diff
@@ -33,8 +33,28 @@ def _on_post(project, trigger):
         ):
             schedule_invalidate_project_config(project_id=project.id, trigger=trigger)
 
+    def _subscribe_project_to_alert_rule(project, release, trigger):
+        """
+        TODO: potentially enable custom query_extra to be passed on ReleaseProject creation (on release/deploy)
+        """
+        from django.utils import timezone
+
+        from sentry.incidents.models import AlertRule
+        from sentry.incidents.utils.types import AlertRuleActivationConditionType
+
+        query_extra = f"release:{release.version} AND event.timestamp:>{timezone.now().isoformat()}"
+        AlertRule.objects.conditionally_subscribe_project_to_alert_rules(
+            project=project,
+            activation_condition=AlertRuleActivationConditionType.RELEASE_CREATION,
+            query_extra=query_extra,
+            trigger=trigger,
+        )
+
     def post_save(self, instance, **kwargs):
         self._on_post(project=instance.project, trigger="releaseproject.post_save")
+        self._subscribe_project_to_alert_rule(
+            project=instance.project, release=instance.release, trigger="releaseproject.post_save"
+        )
```

Comment:
I assume you want this only on creation right? You can check `kwargs["created"]` for that

Reply:
aha perfect, thank you!